### PR TITLE
[CARBONDATA-2530][MV] Fix wrong data displayed when parent table data are loaded

### PR DESCRIPTION
**Pull request by xubo245**
This PR fixed the  wrong data displayed when parent table data are loaded after rebuild for datamap, including insert into and load

 - [x] Any interfaces changed?
 No
 - [x] Any backward compatibility impacted?
 No
 - [x] Document update required?
No
 - [x] Testing done
      add test case for it
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
No

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6587/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5411/

**Comment by xubo245** [comment]
@ravipesala @jackylk Please review it.

File: datamap/mv/core/src/main/scala/org/apache/carbondata/mv/datamap/MVAnalyzerRule.scala [hunk 1]
```diff
@@ -75,6 +77,13 @@ class MVAnalyzerRule(sparkSession: SparkSession) extends Rule[LogicalPlan] {
         plan
       }
     } else {
+      if (catalog != null && (plan.isInstanceOf[InsertIntoCarbonTable]
```
Comment (jackylk) [inline datamap/mv/core/src/main/scala/org/apache/carbondata/mv/datamap/MVAnalyzerRule.scala]: please move `(plan.isInstanceOf[InsertIntoCarbonTable]` to next line

File: datamap/mv/core/src/main/scala/org/apache/carbondata/mv/datamap/MVAnalyzerRule.scala [hunk 2]
```diff
@@ -75,6 +77,13 @@ class MVAnalyzerRule(sparkSession: SparkSession) extends Rule[LogicalPlan] {
...
+        || plan.isInstanceOf[CarbonLoadDataCommand])) {
+        val allSchema = catalog.asInstanceOf[SummaryDatasetCatalog].listAllSchema()
+        for (schema <- allSchema) {
```
Comment (jackylk) [inline datamap/mv/core/src/main/scala/org/apache/carbondata/mv/datamap/MVAnalyzerRule.scala]: use `foreach` instead of `for` which is faster in Scala

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7104/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5880/

**Comment by ravipesala** [comment]
@xubo245 Please close it, it is handled in https://github.com/apache/carbondata/pull/2474

**Comment by xubo245** [comment]
@ravipesala ok

**Pull request closed without merging**